crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -433,32 +433,40 @@ impl Analyzer<'_, '_> {
             is_valid_lhs(&e.left).report(&mut analyzer.storage);
 
             let mut errors = Errors::default();
-            let will_skip_this = if let Some(ty) = type_ann {
+            // skip this flag for arrow function, static method etc.
+            let mut left_function_declare_not_this_type = false;
+            let right_function_declared_this = if let box RExpr::Fn(RFnExpr {
+                function: box stc_ts_ast_rnode::RFunction { params, .. },
+                ..
+            }) = &e.right
+            {
+                if let [RParam {
+                    pat: RPat::Ident(RBindingIdent { id, .. }),
+                    ..
+                }, ..] = &params[..]
+                {
+                    id.sym == js_word!("this")
+                } else {
+                    false
+                }
+            } else {
+                false
+            };
+
+            let rhs_is_arrow = matches!(e.right, box RExpr::Arrow(RArrowExpr { .. }));
```

Comment:
Same here. I think we should handle `(() => {})` too

Reply:
I fixed it [`2d2473f` (#736)](https://github.com/dudykr/stc/pull/736/commits/2d2473f63f3026baf7fe6e50ceaf834818ed801b)